core/src/test/scala/unit/kafka/tools/ConsoleConsumerTest.scala

```diff
@@ -274,9 +334,35 @@ class ConsoleConsumerTest {
     // reset at the end
     try {
       val config = new ConsoleConsumer.ConsumerConfig(args)
-      val consumerProperties = ConsoleConsumer.getNewConsumerProps(config)
-    } finally
-    {
+      ConsoleConsumer.getNewConsumerProps(config)
+    } finally {
+      Exit.resetExitProcedure()
+    }
+
+    // Should have thrown an exception before here, if we reach this line we can fail the test
+    fail()
+  }
+
+  @Test(expected = classOf[IllegalArgumentException])
+  def shouldExitOnInvalidConfigWithAutoOffsetResetAndConflictingFromBeginningOldConsumer() {
+
+    // Override exit procedure to throw an exception instead of exiting, so we can catch the exit
+    // properly for this test case
+    Exit.setExitProcedure((_, message) => throw new IllegalArgumentException(message.orNull))
+
+    //Given
+    val args: Array[String] = Array(
+      "--zookeeper", "localhost:2181",
+      "--topic", "test",
+      "--consumer-property", "auto.offset.reset=largest",
+      "--from-beginning")
+
+    // Enclose test calls in try-finally to ensure the exit procedure is
+    // reset at the end
```

Comment:
I think we don't need this comment. It doesn't add much over the code (a try/finally is done so that the code in `finally` is always run`).